ops/errors.go

```diff
@@ -38,3 +38,6 @@ const AxisOutOfRangeErrTemplate = "axis argument must be in the range -%d <= x <
 // AxesNotAllInRangeErrTemplate is used to format an error when not all indices
 // are within a given range.
 const AxesNotAllInRangeErrTemplate = "all indices entries must be in the range -%d <= x < %d"
+
+// UnsupportedDTypeError is used when the DType of a tensor is not supported.
+const UnsupportedDtypeErrTemplate = "dtype %v is not supported for operator %v"
```

Comment:
With the new linter we don't allow for dynamic errors so instead of using these `consts` and returning the errors with `fmt.Errorf` we should either use a custom error type like :

```go

type UnsupportedDataTypeError struct {
    operator Operator 
    dataType string 
}

func (u UnsuportedDTypeError) Error() string {
      return fmt.Sprintf("dtype %v not supported for %s", u.dataType, u.operator.String()) 
}

func ErrUnsuportedDataType(dataType string, operator Operator) error {
     return UnsuportedDataTypeError{
          operator: operator, 
          dataType: dataType,
     }
} 

// Or we could also doe 

var UnsuportedDataTypeBaseError = errors.New("unsuported dtype") 

func ErrUnsuportedDataType(dataType string, operator string) error {
       return fmt.Errorf("%w: for data type %s, operator %s", UnsuportedDataTypeBaseError, dataType, operator) 
} 

```

With these methods the user is able to assert for certain error types so we should adopt this method.

Reply:
As discussed with @wipsel , we will fix this in https://github.com/AdvancedClimateSystems/gonnx/issues/169